aws/ec2metadata/token_provider.go

```diff
@@ -49,23 +54,21 @@ func (t *tokenProvider) fetchTokenHandler(r *request.Request) {
 	output, err := t.client.getToken(r.Context(), t.configuredTTL)
 
 	if err != nil {
+		// only attempt fallback to insecure data flow if IMDSv1 is enabled
+		if !t.fallbackEnabled() {
+			r.Error = awserr.New("EC2MetadataError", "failed to get IMDS token and fallback is disabled", err)
```

Comment:
nit: i think this error message could have specified what version of IMDS failed and what version its falling back to. so like:
```
failed to get IMDSv2 token and fallback to IMDSv1 is disabled
```